raft.c

```diff
@@ -2418,19 +2416,18 @@ void replaceShardGroups(RedisRaftCtx *rr, raft_entry_t *entry)
         RedisModule_Assert(endptr == nl);
         payload = nl + 1;
 
-        ShardGroup sg;
-        if (ShardGroupDeserialize(payload, payload_len, &sg) != RR_OK) {
+        ShardGroup *sg = ShardGroupCreate();
```

Comment:
Similar issue here, `sg` will leak on early return if I'm not mistaken.

Reply:
fixed, by moving the ShardGroupCreate() into ShardGroupDeserialize and returning it via normal return conventions.  instead of RR_OK/RR_ERR, we now test that the it's not NULL.